File: app/src/main/java/com/example/doan/Models/Order.java
package com.example.doan.Models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class Order implements Serializable {

    @SerializedName("id")
    private int id;

    // Bảng DB không có order_number, dùng ID làm mã đơn
    @SerializedName("order_number")
    private String orderNumber;

    // Ánh xạ từ created_at trong DB
    @SerializedName("created_at")
    private String date;

    @SerializedName("status")
    private String status;

    // Ánh xạ từ total_price trong DB
    @SerializedName("total_price")
    private double totalAmount;
    
    @SerializedName("final_price")
    private double finalPrice;
    
    @SerializedName("address")
    private String address;

    // Customer info
    @SerializedName("customer_name")
    private String customerName;
    
    @SerializedName("customer_phone")
    private String customerPhone;
    
    // Payment info
    @SerializedName("payment_status")
    private String paymentStatus;
    
    @SerializedName("payment_method")
    private String paymentMethod;
    
    // Timestamp
    @SerializedName("createdAt")
    private String createdAt;
    
    @SerializedName("updatedAt")
    private String updatedAt;

    public Order(int id, String orderNumber, String date, String status, double totalAmount) {
        this.id = id;
        this.orderNumber = orderNumber;
        this.date = date;
        this.status = status;
        this.totalAmount = totalAmount;
    }

    public Order() {
    }

    public int getId() {
        return id;
    }

    public String getOrderNumber() {
        // Nếu server không trả về order_number, dùng ID thay thế
        if (orderNumber == null || orderNumber.isEmpty()) {
            return String.valueOf(id);
        }
        return orderNumber;
    }

    public String getDate() {
        return date;
    }

    public String getStatus() {
        return status;
    }

    public double getTotalAmount() {
        return totalAmount;
    }
    
    public double getFinalPrice() {
        return finalPrice;
    }
    
    public String getAddress() {
        return address;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setOrderNumber(String orderNumber) {
        this.orderNumber = orderNumber;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setTotalAmount(double totalAmount) {
        this.totalAmount = totalAmount;
    }
    
    public String getCustomerName() {
        return customerName;
    }
    
    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }
    
    public String getCustomerPhone() {
        return customerPhone;
    }
    
    public void setCustomerPhone(String customerPhone) {
        this.customerPhone = customerPhone;
    }
    
    public String getPaymentStatus() {
        return paymentStatus;
    }
    
    public void setPaymentStatus(String paymentStatus) {
        this.paymentStatus = paymentStatus;
    }
    
    public String getPaymentMethod() {
        return paymentMethod;
    }
    
    public void setPaymentMethod(String paymentMethod) {
        this.paymentMethod = paymentMethod;
    }
    
    public String getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }
    
    public String getUpdatedAt() {
        return updatedAt;
    }
    
    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: app/src/main/java/com/example/doan/Models/Order.kt
package com.example.doan.Models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * FIX C7: Cập nhật SerializedName để khớp với Backend OrderDto
 * FIX C4: Chuyển từ Serializable sang Parcelable để tránh crash trên một số thiết bị
 * Backend trả về: id, userId, userName, storeId, storeName, type, address, 
 *                 pickupTime, status, totalPrice, discount, finalPrice, 
 *                 paymentMethod, promotionCode, items, createdAt, updatedAt
 */
@Parcelize
data class Order(
    @SerializedName("id")
    var id: Int = 0,
    
    @SerializedName("userId")
    var userId: Long? = null,
    
    @SerializedName("userName")
    var userName: String? = null,
    
    @SerializedName("storeId")
    var storeId: Long? = null,
    
    @SerializedName("storeName")
    var storeName: String? = null,
    
    @SerializedName("type")
    var type: String? = null,
    
    @SerializedName("status")
    var status: String? = null,
    
    // FIX C7: Backend trả về "totalPrice" không phải "total_price"
    @SerializedName("totalPrice")
    var totalPrice: Double = 0.0,
    
    @SerializedName("discount")
    var discount: Double = 0.0,
    
    // FIX C7: Backend trả về "finalPrice" không phải "final_price"
    @SerializedName("finalPrice")
    var finalPrice: Double = 0.0,
    
    @SerializedName("address")
    var address: String? = null,
    
    @SerializedName("pickupTime")
    var pickupTime: String? = null,
    
    // FIX C7: Backend trả về "paymentMethod" không phải "payment_method"
    @SerializedName("paymentMethod")
    var paymentMethod: String? = null,
    
    @SerializedName("promotionCode")
    var promotionCode: String? = null,
    
    // FIX C7: Backend trả về "createdAt" (đã đúng)
    @SerializedName("createdAt")
    var createdAt: String? = null,
    
    @SerializedName("updatedAt")
    var updatedAt: String? = null,
    
    @SerializedName("items")
    var items: List<OrderItem>? = null
) : Parcelable {
    
    // Backward compatibility - giữ lại các property cũ
    val totalAmount: Double get() = totalPrice
    val customerName: String? get() = userName
    
    fun getDisplayOrderNumber(): String {
        return "#${id}"
    }
    
    fun getDisplayStatus(): String {
        return when (status) {
            "PENDING" -> "Chờ xử lý"
            "MAKING" -> "Đang pha chế"
            "SHIPPING" -> "Đang giao"
            "READY" -> "Sẵn sàng"
            "DONE" -> "Hoàn thành"
            "CANCELED" -> "Đã hủy"
            else -> status ?: "Không xác định"
        }
    }
}
